File: programming/lab2/src/main/java/moves/status/SwordsDance.java
package moves.status;

import ru.ifmo.se.pokemon.*;

/**
 * Swords Dance raises the user's Attack by two stages.
 */
public class SwordsDance extends StatusMove {
    public SwordsDance() {
        super(Type.NORMAL, 0, 100);
    }

    @Override
    protected String describe() {
        return "использует Swords Dance";
    }
    @Override
    protected boolean checkAccuracy(Pokemon pokemon, Pokemon pokemon1) {
        return true;
    }

    @Override
    protected void applySelfEffects(Pokemon pokemon) {
        pokemon.addEffect(new Effect().stat(Stat.ATTACK, 2).turns(-1));
    }


}

File: programming/lab2/app/src/main/kotlin/org/itmo/moves/status/SwordsDance.kt
package org.itmo.moves.status

import ru.ifmo.se.pokemon.*

class SwordsDance : StatusMove(Type.NORMAL, 0.0, .0) {
    override fun applySelfEffects(p0: Pokemon) {
        p0.addEffect(Effect().stat(Stat.ATTACK, 2))
    }
}